src/tm_devices/drivers/pi/scopes/tekscope_2k/tekscope_2k.py

```diff
@@ -18,10 +24,117 @@ class TekScope2k(Scope, ABC):
     # Properties
     ################################################################################################
 
+    @cached_property
+    def total_channels(self) -> int:
+        """Return the total number of channels (all types)."""
+        return int(self.model[6])
+
     ################################################################################################
     # Public Methods
     ################################################################################################
 
+    def waveforms_displayed(self) -> List[str]:
+        """Returns list of waveforms currently displayed on oscilloscope screen.
+
+        Args:
```

Comment:
remove this section, since there are no args

Reply:
Args section removed.